frame/democracy/src/lib.rs

```diff
@@ -1191,7 +1191,10 @@ pub mod pallet {
 		}
 
 		/// Enact a proposal from a referendum. For now we just make the weight be the maximum.
-		#[pallet::weight(T::BlockWeights::get().max_block)]
+		// NOTE: This is badbadbad. The entire mechanism should be scrapped in favour of using
+		// Scheduler throughout. However we should move all Democracy over to Gov2 anyway, so may
+		// as well just deprecate this pallet.
+		#[pallet::weight(T::BlockWeights::get().max_block / 5)]
```

Comment:
basically the scheduler has been changed to become much safer on parachains. it used to execute a task if:
- there was enough weight left for the amount the task may need;
- it was the first task we're trying to execute;
- it is of a "hard-deadline" priority.

the last two were potentially useful features practically speaking, but they were dangerous as they could make a fundamentally overweight task mandatory to execute and thus risk the block being unimportable and the chain stalling. 

so they're gone now. but this means that a number of tests fail since we no longer guarantee that at least one task will execute every block and some of the test infrastructure was relying on this. in particular, the Democracy pallet dispatches its proposals through scheduling a re-dispatcher function which then makes use of Democracy's `Preimages` map. this re-dispatcher function cannot determine the weight of the inner call at the time of weight-calculation (since it is behind the `Preimages` map) so it was just assuming that it would take up the entire block weight. this used to work with the old Scheduler logic but now it will just fail.

instead of doing this, we really want to remove the re-dispatcher altogether as well as Democracy's `Preimages` map and just schedule the call directly. it's in the PR's TODOs.